### PR TITLE
SPR-16425 - Remove square brackets around stringified collections

### PR DESCRIPTION
**Pull request by logart**

### CODE REVIEW

**Comment by pivotal-issuemaster** [comment]
@logart Please sign the [Contributor License Agreement](https://cla.pivotal.io/sign/spring?repositoryId=spring-projects/spring-framework&pullRequestId=1655)!

[Click here](https://cla.pivotal.io/sync/spring?repositoryId=spring-projects/spring-framework&pullRequestId=1655) to manually synchronize the status of this Pull Request.

See the [FAQ](https://cla.pivotal.io/about) for frequently asked questions.

**Comment by pivotal-issuemaster** [comment]
@logart Thank you for signing the [Contributor License Agreement](https://cla.pivotal.io/sign/spring?repositoryId=spring-projects/spring-framework&pullRequestId=1655)!

File: spring-web/src/main/java/org/springframework/web/util/UriComponents.java [hunk 1]
```diff
@@ -273,7 +271,15 @@ private static String getVariableName(String match) {
 	}
 
 	private static String getVariableValueAsString(@Nullable Object variableValue) {
-		return (variableValue != null ? variableValue.toString() : "");
+		if (variableValue != null) {
```
Comment (fmcejudo) [inline spring-web/src/main/java/org/springframework/web/util/UriComponents.java]: might we have this as `if(variableValue == null) {  return ""; }`? and afterward keep evaluating the variable... so, I had in mind something like:
` if(Collection.class.isAssignableFrom(variableValue.getClass())){
      return ((Collection) variableValue).stream().map(Object::toString).collect(Collectors.joining(","));
} else {
   return variableValue.toString();
}`

so, we are safe from a NullPointerException and avoiding nested blocks, what makes the code cleaner, and using a little bit more streams collections features. 

https://docs.oracle.com/javase/8/docs/api/java/util/stream/Collectors.html#joining-java.lang.CharSequence

What do you think?

**Comment by fmcejudo** [comment]
Also, because there are two commits for only one file and purpose, you might want to squash them into one with rebase -i HEAD~2

https://stackoverflow.com/questions/2563632/how-can-i-merge-two-commits-into-one

**Comment by logart** [comment]
Hello, sorry for long response.
Sure, I will do it today.

**Comment by rstoyanchev** [comment]
Before you do that, please check my comment under SPR-16425.

**Comment by rstoyanchev** [comment]
Closing as the related issue has also been closed.

**Pull request closed without merging**